doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -848,37 +1231,425 @@ The Ltac2 parser can be extended with syntactic notations.
 Abbreviations
 ~~~~~~~~~~~~~
 
-.. cmdv:: Ltac2 Notation @lident := @ltac2_term
+.. cmd:: Ltac2 Notation {| @string | @lident } := @ltac2_expr
+   :name: Ltac2 Notation (abbreviation)
 
-  This command introduces a special kind of notation, called an abbreviation,
-  that is designed so that it does not add any parsing rules. It is similar in
-  spirit to Coq abbreviations, insofar as its main purpose is to give an
-  absolute name to a piece of pure syntax, which can be transparently referred to
-  by this name as if it were a proper definition.
+   Introduces a special kind of notation, called an abbreviation,
+   that does not add any parsing rules. It is similar in
+   spirit to Coq abbreviations (see :cmd:`Notation (abbreviation)`,
+   insofar as its main purpose is to give an
+   absolute name to a piece of pure syntax, which can be transparently referred to
+   by this name as if it were a proper definition.
 
-  The abbreviation can then be manipulated just as a normal Ltac2 definition,
-  except that it is expanded at internalization time into the given expression.
-  Furthermore, in order to make this kind of construction useful in practice in
-  an effectful language such as Ltac2, any syntactic argument to an abbreviation
-  is thunked on-the-fly during its expansion.
+   The abbreviation can then be manipulated just like a normal Ltac2 definition,
+   except that it is expanded at internalization time into the given expression.
+   Furthermore, in order to make this kind of construction useful in practice in
+   an effectful language such as Ltac2, any syntactic argument to an abbreviation
+   is thunked on-the-fly during its expansion.
 
-For instance, suppose that we define the following.
+   For instance, suppose that we define the following.
 
-:n:`Ltac2 Notation foo := fun x => x ().`
+   :n:`Ltac2 Notation foo := fun x => x ().`
 
-Then we have the following expansion at internalization time.
+   Then we have the following expansion at internalization time.
 
-:n:`foo 0 ↦ (fun x => x ()) (fun _ => 0)`
+   :n:`foo 0 ↦ (fun x => x ()) (fun _ => 0)`
 
-Note that abbreviations are not typechecked at all, and may result in typing
-errors after expansion.
+   Note that abbreviations are not type checked at all, and may result in typing
+   errors after expansion.
+
+.. _defining_tactics:
+
+Defining tactics
+~~~~~~~~~~~~~~~~
+
+All tactics used in Ltac2 must defined through notations.  These include built-in tactics (those
```

Comment:
```suggestion
All tactics used in Ltac2 must be defined through notations.  These include built-in tactics (those
```
This seems false, though, as you can define tactics with `Ltac2 foo := ...`.  Or, at least, it seems unclear what is meant to a newcomer.

Reply:
Indeed, nothing forces you to go through notations and it's a good thing. It'd be terrible to have to...

Reply:
I think the intended meaning was something like "you can't just call Ltac1 tactics in Ltac2 without defining notations for them or wrapping in ltac1:(...)"?